go/vt/vtgate/safe_session.go

```diff
@@ -571,6 +571,22 @@ func (session *SafeSession) TimeZone() *time.Location {
 	loc, _ := datetime.ParseTimeZone(tz)
 	return loc
 }
+func (session *SafeSession) ForeignKeyChecks() *string {
+	session.mu.Lock()
+	fkVal, ok := session.SystemVariables["foreign_key_checks"]
+	session.mu.Unlock()
+
+	if !ok {
+		return nil
+	}
+	fkCheck := "0"
+	switch strings.ToLower(fkVal) {
+	case "off", "0":
+		return &fkCheck
+	}
+	fkCheck = "1"
```

Comment:
This doesn't look right. People can set the environment variable to an incorrect value, in which case we should return nil. (That's what MySQL does). So we should check for "1" and "on" and for everything else return nil.

Reply:
probably a reasonable compromise.

Reply:
What do you mean by a reasonable compromise? We should just return nil when the value isn't "1" or "on". We don't want to reject those settings. They just won't do anything

Reply:
a reasonable compromise is what you suggested as if the value is not recognized then ignore such system settings.
Ideally, we should fail when we receive the `set` statement for unknown values.